src/vanilla/atom.ts

```diff
@@ -23,7 +23,7 @@ type Write<Args extends unknown[], Result> = (
   ...args: Args
 ) => Result
 
-type WithInitialValue<Value> = {
+export type WithInitialValue<Value> = {
```

Comment:
This type is intentionally private. Please define it in atomWithReset.ts.
`init` property is an internal property too.